### DISCUSSION TITLE
Compile Error with clang '-mfp16-format=ieee'

### DISCUSSION THREAD

**Discussion by mbrunnen** [zephyrproject-rtos/zephyr#55562] [General]
I know that `llvm` is not supported right now for ARM targets, but i think this could be an easy fix.

So, I have set the environment variable `ZEPHYR_TOOLCHAIN_VARIANT=llvm` and this is the error:

```
clang: error: unknown argument: '-mfp16-format=ieee'
```

I think it is because `clang/target.cmake` includes `gcc/target_arm.cmake`:

https://github.com/zephyrproject-rtos/zephyr/blob/96c8d16986b64a36c24ae8ab0f0d82df22fa60b5/cmake/compiler/clang/target.cmake#L32

`gcc/target_arm.cmake` sets `-mfp16-format=ieee` which is not yet supported by llvm.
But as far as I understand, it is enabled by llvm by default:

https://github.com/llvm/llvm-project/blob/b189218d4422ba3f4676fa1191961df8e4732783/llvm/lib/Target/ARM/ARMAsmPrinter.cpp#L734
```
  // FIXME: To support emitting this build attribute as GCC does, the
  // -mfp16-format option and associated plumbing must be
  // supported. For now the __fp16 type is exposed by default, so this
  // attribute should be emitted with value 1.
  ATS.emitAttribute(ARMBuildAttrs::ABI_FP_16bit_format,
                    ARMBuildAttrs::FP16FormatIEEE);
```

**Comment by henrikbrixandersen** [comment]
Please use our bug template when reporting bugs. You need to edit this issue to include the needed information from https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by awojasinski** [comment]
Issue has been fixed in https://github.com/zephyrproject-rtos/zephyr/commit/f16c4324a7e28c929f7226dd06fef87d951431c9#diff-fc6ea5b3151b6f25bc3324263d55cb9572163281048841fbb1a5d76f5341fb4dL32
